src/components/TheNav.vue

```diff
@@ -42,10 +42,11 @@
                     >
                         <ul class="nav-items">
                             <li
-                                @click="toggleNav(); item.function"
+                                @click="toggleNav(); item.callback()"
```

Comment:
instead of adding a callback as separate listener you can pass it to the click handler (`toggleNav`) as an argument

Reply:
I don't think I can because this callback is actually supposed to sign out the user (it's not working right know, though) and the toggleNav function not only is triggered when a user clicks on any of the nav items but also when this is happening on mobile devices.

So, if I did that:
* on mobile, a click on each of the items would sign out the user
* on desktop the nav isn't supposed to be toggled in the first place

Reply:
I understand, but you can easily compose it with toggleNav:
```js
toggleNav(cb) {
      this.isNavShown = !this.isNavShown

      if (cb) {
        cb()
      }
    }
```

Reply:
and then use it in the template

```jsx
@click="() => toggleNav(item.callback)"

Reply:
So, instead of ` @click="toggleNav(); item.callback()"` it should be `toggleNav(item.callback)`? This way, the `toggleNav` will fire for each of the items but this `callback` function remains working only for one item?

Reply:
Only for item with defined (and truthy) callback

Reply:
Alright, I'm still tied to the idea that I always have to provide an argument when my function accepts a parameter so I didn't  think like that.

I also have some doubts of opening those brackets (to provide an argument) because it seems to me like opened brackets work only when more than one functions are called; in every other case I should write just `@click=toggleNav`. But I'll try to implement this and we'll see. I probably just don't understand very well how it works in Vue.

Reply:
this is the same as you'd provide a callback to addEventListener('click', callback)

Reply:
You have to provide a reference to a function that will be called on that particular event

Reply:
I did it.... somehow 😛 It works but it does trigger some errors, don't know it it's an issue...